yt_dlp/extractor/getcourseru.py

```diff
@@ -0,0 +1,181 @@
+        'info_dict': {
+            'id': '3059742',
+            'display_id': '3video_1',
+            'title': 'Промоуроки Академии МЕЛ',
+        },
+        'playlist_count': 1,
+        'playlist': [{
+            'info_dict': {
+                'id': '513573381',
+                'ext': 'mp4',
+                'title': 'Промоуроки Академии МЕЛ',
+                'thumbnail': 'https://preview-htz.kinescopecdn.net/preview/190bdf93f1b29735309853a7a19e24b3/preview.jpg?version=1702370546&host=vh-80',
+                'duration': 1693
+            },
+        }]
+    }, {
+        'url': 'https://academymel.getcourse.ru/pl/teach/control/lesson/view?id=319141781&editMode=0',
+        'info_dict': {
+            'id': '319141781',
+            'title': '1. Разминка у стены',
+        },
+        'playlist_count': 1,
+        'playlist': [{
+            'info_dict': {
+                'id': '4919601',
+                'ext': 'mp4',
+                'title': '1. Разминка у стены',
+                'thumbnail': 'https://preview-htz.vhcdn.com/preview/5a521788e7dc25b4f70c3dff6512d90e/preview.jpg?version=1703223532&host=vh-81',
+                'duration': 704
+            },
+        }],
+        'skip': 'paid lesson'
+    }, {
+        'url': 'https://manibeauty.getcourse.ru/pl/teach/control/lesson/view?id=272499894',
+        'info_dict': {
+            'id': '272499894',
+            'title': 'Мотивация к тренировкам',
+        },
+        'playlist_count': 1,
+        'playlist': [{
+            'info_dict': {
+                'id': '4242723',
+                'ext': 'mp4',
+                'title': 'Мотивация к тренировкам',
+                'thumbnail': 'https://preview-htz.vhcdn.com/preview/70ed5b9f489dd03b4aff55bfdff71a26/preview.jpg?version=1685115787&host=vh-71',
+                'duration': 30
+            },
+        }],
+        'skip': 'paid lesson'
+    }, {
+        'url': 'https://gaismasmandalas.getcourse.io/ATLAUTSEVBUT',
+        'only_matching': True,
+    }]
+    _DOMAINS = [
+        'academymel.online',
+        'marafon.mani-beauty.com',
+    ]
+    _BASE_URL_RE = rf'https?://(?:(?!player02\.)[^.]+\.getcourse\.(?:ru|io)|{"|".join(map(re.escape, _DOMAINS))})'
+    _VALID_URL = [
+        rf'{_BASE_URL_RE}/(?!pl/|teach/)(?P<id>[^?#]+)',
+        rf'{_BASE_URL_RE}/(:?pl/)?teach/control/lesson/view\?(?:[^#]+&)?id=(?P<id>\d+)',
+    ]
+
+    def _login(self, hostname, username, password):
+        if self._get_cookies(f'https://{hostname}').get('PHPSESSID5'):
+            return
+        login_url = f'https://{hostname}{self._LOGIN_URL_PATH}'
+
+        webpage = self._download_webpage(login_url, None)
+        xdget_id = self._html_search_regex(
+            r'<form[^>]*class="[^"]*state-login[^"]*"[^>]*data-xdget-id="([^"]+)"',
+            webpage, 'xdgetId')
+
+        simple_sign = self._html_search_regex(
+            r'window.requestSimpleSign\s*=\s*"([\da-f]+)"',
+            webpage, 'simple sign')
```

Comment:
```suggestion
            r'window.requestSimpleSign\s*=\s*"([\da-f]+)"', webpage, 'simple sign')
```
or even better, this could be inlined below